### PR TITLE
zmq: add variants; zmq-devel: update to current upstream; both: support tests

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

I made everything into separate commits, so it should be self-explanatory.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

Tests in Rosetta:
```
--->  Testing zmq-devel
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_zmq/zmq-devel/work/build" && ctest test 
Test project /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_zmq/zmq-devel/work/build
        Start   1: test_ancillaries
  1/102 Test   #1: test_ancillaries .................   Passed    0.03 sec
        Start   2: test_system
  2/102 Test   #2: test_system ......................   Passed    0.06 sec
        Start   3: test_pair_inproc
  3/102 Test   #3: test_pair_inproc .................   Passed    0.05 sec
        Start   4: test_pair_tcp
  4/102 Test   #4: test_pair_tcp ....................   Passed    0.07 sec
        Start   5: test_reqrep_inproc
  5/102 Test   #5: test_reqrep_inproc ...............   Passed    0.04 sec
        Start   6: test_reqrep_tcp
  6/102 Test   #6: test_reqrep_tcp ..................   Passed    0.18 sec
        Start   7: test_hwm
  7/102 Test   #7: test_hwm .........................   Passed    2.15 sec
        Start   8: test_hwm_pubsub
  8/102 Test   #8: test_hwm_pubsub ..................***Failed    3.54 sec
        Start   9: test_reqrep_device
  9/102 Test   #9: test_reqrep_device ...............   Passed    0.07 sec
        Start  10: test_sub_forward
 10/102 Test  #10: test_sub_forward .................   Passed    0.37 sec
        Start  11: test_invalid_rep
 11/102 Test  #11: test_invalid_rep .................   Passed    0.04 sec
        Start  12: test_msg_flags
 12/102 Test  #12: test_msg_flags ...................   Passed    0.04 sec
        Start  13: test_msg_ffn
 13/102 Test  #13: test_msg_ffn .....................   Passed    1.26 sec
        Start  14: test_connect_resolve
 14/102 Test  #14: test_connect_resolve .............   Passed    0.07 sec
        Start  15: test_immediate
 15/102 Test  #15: test_immediate ...................   Passed    3.93 sec
        Start  16: test_last_endpoint
 16/102 Test  #16: test_last_endpoint ...............   Passed    0.05 sec
        Start  17: test_term_endpoint
 17/102 Test  #17: test_term_endpoint ...............   Passed    0.69 sec
        Start  18: test_router_mandatory
 18/102 Test  #18: test_router_mandatory ............   Passed    0.06 sec
        Start  19: test_probe_router
 19/102 Test  #19: test_probe_router ................   Passed    0.08 sec
        Start  20: test_stream
 20/102 Test  #20: test_stream ......................   Passed    0.08 sec
        Start  21: test_stream_empty
 21/102 Test  #21: test_stream_empty ................   Passed    0.06 sec
        Start  22: test_stream_disconnect
 22/102 Test  #22: test_stream_disconnect ...........   Passed    0.06 sec
        Start  23: test_disconnect_inproc
 23/102 Test  #23: test_disconnect_inproc ...........   Passed    0.66 sec
        Start  24: test_unbind_wildcard
 24/102 Test  #24: test_unbind_wildcard .............***Failed    0.10 sec
        Start  25: test_ctx_options
 25/102 Test  #25: test_ctx_options .................   Passed    0.04 sec
        Start  26: test_ctx_destroy
 26/102 Test  #26: test_ctx_destroy .................   Passed    0.35 sec
        Start  27: test_security_no_zap_handler
 27/102 Test  #27: test_security_no_zap_handler .....   Passed    0.06 sec
        Start  28: test_security_null
 28/102 Test  #28: test_security_null ...............   Passed    1.35 sec
        Start  29: test_security_plain
 29/102 Test  #29: test_security_plain ..............   Passed    2.35 sec
        Start  30: test_security_zap
 30/102 Test  #30: test_security_zap ................   Passed   34.64 sec
        Start  31: test_iov
 31/102 Test  #31: test_iov .........................   Passed    0.35 sec
        Start  32: test_spec_req
 32/102 Test  #32: test_spec_req ....................   Passed    3.36 sec
        Start  33: test_spec_rep
 33/102 Test  #33: test_spec_rep ....................   Passed    0.69 sec
        Start  34: test_spec_dealer
 34/102 Test  #34: test_spec_dealer .................   Passed    1.82 sec
        Start  35: test_spec_router
 35/102 Test  #35: test_spec_router .................   Passed    1.27 sec
        Start  36: test_spec_pushpull
 36/102 Test  #36: test_spec_pushpull ...............   Passed    6.36 sec
        Start  37: test_req_correlate
 37/102 Test  #37: test_req_correlate ...............   Passed    0.06 sec
        Start  38: test_req_relaxed
 38/102 Test  #38: test_req_relaxed .................   Passed    6.85 sec
        Start  39: test_conflate
 39/102 Test  #39: test_conflate ....................   Passed    0.35 sec
        Start  40: test_inproc_connect
 40/102 Test  #40: test_inproc_connect ..............   Passed    0.50 sec
        Start  41: test_issue_566
 41/102 Test  #41: test_issue_566 ...................   Passed    0.62 sec
        Start  42: test_shutdown_stress
 42/102 Test  #42: test_shutdown_stress .............   Passed    5.79 sec
        Start  43: test_timeo
 43/102 Test  #43: test_timeo .......................   Passed    0.30 sec
        Start  44: test_many_sockets
 44/102 Test  #44: test_many_sockets ................   Passed    0.76 sec
        Start  45: test_diffserv
 45/102 Test  #45: test_diffserv ....................***Exception: Illegal  0.13 sec
        Start  46: test_connect_rid
 46/102 Test  #46: test_connect_rid .................   Passed    0.71 sec
        Start  47: test_xpub_nodrop
 47/102 Test  #47: test_xpub_nodrop .................   Passed    0.36 sec
        Start  48: test_pub_invert_matching
 48/102 Test  #48: test_pub_invert_matching .........   Passed    1.25 sec
        Start  49: test_setsockopt
 49/102 Test  #49: test_setsockopt ..................   Passed    0.05 sec
        Start  50: test_sockopt_hwm
 50/102 Test  #50: test_sockopt_hwm .................   Passed    0.96 sec
        Start  51: test_heartbeats
 51/102 Test  #51: test_heartbeats ..................   Passed   36.56 sec
        Start  52: test_atomics
 52/102 Test  #52: test_atomics .....................   Passed    0.03 sec
        Start  53: test_bind_src_address
 53/102 Test  #53: test_bind_src_address ............   Passed    0.05 sec
        Start  54: test_capabilities
 54/102 Test  #54: test_capabilities ................   Passed    0.03 sec
        Start  55: test_metadata
 55/102 Test  #55: test_metadata ....................   Passed    0.37 sec
        Start  56: test_router_handover
 56/102 Test  #56: test_router_handover .............   Passed    0.99 sec
        Start  57: test_srcfd
 57/102 Test  #57: test_srcfd .......................   Passed    0.36 sec
        Start  58: test_stream_timeout
 58/102 Test  #58: test_stream_timeout ..............   Passed    0.28 sec
        Start  59: test_xpub_manual
 59/102 Test  #59: test_xpub_manual .................   Passed    4.02 sec
        Start  60: test_xpub_welcome_msg
 60/102 Test  #60: test_xpub_welcome_msg ............   Passed    0.04 sec
        Start  61: test_xpub_verbose
 61/102 Test  #61: test_xpub_verbose ................   Passed    0.07 sec
        Start  62: test_base85
 62/102 Test  #62: test_base85 ......................   Passed    0.04 sec
        Start  63: test_bind_after_connect_tcp
 63/102 Test  #63: test_bind_after_connect_tcp ......   Passed    0.07 sec
        Start  64: test_sodium
 64/102 Test  #64: test_sodium ......................   Passed    0.05 sec
        Start  65: test_monitor
 65/102 Test  #65: test_monitor .....................   Passed    0.07 sec
        Start  66: test_socket_null
 66/102 Test  #66: test_socket_null .................   Passed    0.03 sec
        Start  67: test_reconnect_ivl
 67/102 Test  #67: test_reconnect_ivl ...............   Passed    7.64 sec
        Start  68: test_reconnect_options
 68/102 Test  #68: test_reconnect_options ...........   Passed    3.46 sec
        Start  69: test_tcp_accept_filter
 69/102 Test  #69: test_tcp_accept_filter ...........   Passed    1.12 sec
        Start  70: test_mock_pub_sub
 70/102 Test  #70: test_mock_pub_sub ................   Passed    0.09 sec
        Start  71: test_security_gssapi
 71/102 Test  #71: test_security_gssapi .............   Passed    0.03 sec
        Start  72: test_socks
 72/102 Test  #72: test_socks .......................   Passed    3.17 sec
        Start  73: test_connect_null_fuzzer
 73/102 Test  #73: test_connect_null_fuzzer .........***Skipped   0.03 sec
        Start  74: test_bind_null_fuzzer
 74/102 Test  #74: test_bind_null_fuzzer ............***Skipped   0.03 sec
        Start  75: test_connect_fuzzer
 75/102 Test  #75: test_connect_fuzzer ..............***Skipped   0.03 sec
        Start  76: test_bind_fuzzer
 76/102 Test  #76: test_bind_fuzzer .................***Skipped   0.03 sec
        Start  77: test_security_curve
 77/102 Test  #77: test_security_curve ..............   Passed   12.32 sec
        Start  78: test_connect_curve_fuzzer
 78/102 Test  #78: test_connect_curve_fuzzer ........***Skipped   0.03 sec
        Start  79: test_bind_curve_fuzzer
 79/102 Test  #79: test_bind_curve_fuzzer ...........***Skipped   0.03 sec
        Start  80: test_z85_decode_fuzzer
 80/102 Test  #80: test_z85_decode_fuzzer ...........***Skipped   0.03 sec
        Start  81: test_ipc_wildcard
 81/102 Test  #81: test_ipc_wildcard ................   Passed    0.05 sec
        Start  82: test_pair_ipc
 82/102 Test  #82: test_pair_ipc ....................   Passed    0.06 sec
        Start  83: test_reqrep_ipc
 83/102 Test  #83: test_reqrep_ipc ..................   Passed    0.36 sec
        Start  84: test_rebind_ipc
 84/102 Test  #84: test_rebind_ipc ..................   Passed    0.21 sec
        Start  85: test_proxy
 85/102 Test  #85: test_proxy .......................   Passed    1.12 sec
        Start  86: test_proxy_hwm
 86/102 Test  #86: test_proxy_hwm ...................   Passed    3.18 sec
        Start  87: test_proxy_single_socket
 87/102 Test  #87: test_proxy_single_socket .........   Passed    0.07 sec
        Start  88: test_proxy_terminate
 88/102 Test  #88: test_proxy_terminate .............   Passed    0.96 sec
        Start  89: test_getsockopt_memset
 89/102 Test  #89: test_getsockopt_memset ...........   Passed    0.04 sec
        Start  90: test_filter_ipc
 90/102 Test  #90: test_filter_ipc ..................   Passed    0.06 sec
        Start  91: test_stream_exceeds_buffer
 91/102 Test  #91: test_stream_exceeds_buffer .......   Passed    0.05 sec
        Start  92: test_router_mandatory_hwm
 92/102 Test  #92: test_router_mandatory_hwm ........   Passed    1.07 sec
        Start  93: test_use_fd
 93/102 Test  #93: test_use_fd ......................   Passed    0.09 sec
        Start  94: test_zmq_poll_fd
 94/102 Test  #94: test_zmq_poll_fd .................   Passed    0.04 sec
        Start  95: test_fork
 95/102 Test  #95: test_fork ........................   Passed    0.07 sec
        Start  96: unittest_ypipe
 96/102 Test  #96: unittest_ypipe ...................   Passed    0.03 sec
        Start  97: unittest_poller
 97/102 Test  #97: unittest_poller ..................   Passed    0.08 sec
        Start  98: unittest_mtrie
 98/102 Test  #98: unittest_mtrie ...................   Passed    0.04 sec
        Start  99: unittest_ip_resolver
 99/102 Test  #99: unittest_ip_resolver .............***Failed    0.05 sec
        Start 100: unittest_udp_address
100/102 Test #100: unittest_udp_address .............   Passed    0.47 sec
        Start 101: unittest_radix_tree
101/102 Test #101: unittest_radix_tree ..............   Passed    0.03 sec
        Start 102: unittest_curve_encoding
102/102 Test #102: unittest_curve_encoding ..........   Passed    0.08 sec

96% tests passed, 4 tests failed out of 102

Total Test time (real) = 164.75 sec

The following tests did not run:
	 73 - test_connect_null_fuzzer (Skipped)
	 74 - test_bind_null_fuzzer (Skipped)
	 75 - test_connect_fuzzer (Skipped)
	 76 - test_bind_fuzzer (Skipped)
	 78 - test_connect_curve_fuzzer (Skipped)
	 79 - test_bind_curve_fuzzer (Skipped)
	 80 - test_z85_decode_fuzzer (Skipped)

The following tests FAILED:
	  8 - test_hwm_pubsub (Failed)
	 24 - test_unbind_wildcard (Failed)
	 45 - test_diffserv (ILLEGAL)
	 99 - unittest_ip_resolver (Failed)
Errors while running CTest
```

### CODE REVIEW

**Comment by barracuda156** [comment]
I do not get what is going on with checksums now. I will update in few min.

**Comment by michaelld** [comment]
@barracuda156 are you ready for me to start reviewing this PR? Or, some more work to do here?

**Comment by barracuda156** [comment]
> @barracuda156 are you ready for me to start reviewing this PR? Or, some more work to do here?

@michaelld Please review, ready for now. I won’t touch obsolete subports there.

File: devel/zmq/Portfile [hunk 1]
```diff
@@ -99,7 +103,7 @@ subport zmq-devel {
     PortGroup cmake 1.1
 
     github.setup    zeromq libzmq 875c2b15c30af70010c6681539b2195ba391069f
-    version         20221002-[string range ${github.version} 0 7]
+    version         20230201-[string range ${github.version} 0 7]
```
Comment (michaelld) [inline devel/zmq/Portfile:106]: Need to fix this ... the port date will not match the hash date after this! You can either do a rev-bump or update the date, hash, & checksum ... and this latter is what I recommend:
```
    github.setup    zeromq libzmq 8c725093ac4b44a97e6cb64566989ef12b71986c
    version         20230206-[string range ${github.version} 0 7]
    checksums       rmd160  c870f927dc0418a401bbe0847bb8ce0aa3576168 \
                    sha256  4dc235243ae91cbe16680d3408fa16ae777c6f277190f44cd271292ae4a231db \
                    size    939179
    revision        0
```

**Review by michaelld — CHANGES_REQUESTED** [review]
1 change & then I think this is good to go

**Comment by barracuda156** [comment]
> 1 change & then I think this is good to go

@michaelld Will do it today, thank you.

File: devel/zmq/Portfile (hunk 1 above)
Comment (barracuda156) [reply devel/zmq/Portfile]: Fixed and rebased.

**Review by michaelld — APPROVED** [review]
LGTM ; thx!

**Comment by barracuda156** [comment]
> LGTM ; thx!

@michaelld Thank you!

**Comment by michaelld** [comment]
There is some chance that tweaking the C11/C++11 stuff will cause build issues on older OSX. That said, I'm not worried about any OSX/macOS 10.11 or newer, which (in my belief) is the bulk of MP users. Hence, it's worth merging & if/as necessary fixing up post-merge.

**Pull request merged**